CodeChallenge.md

```diff
@@ -0,0 +1,17 @@
+# Blockchain Engineer Code Challenge
+
+Assuming we are designing an on-chain notebook for EVM, where each notebook is an ERC-721 token. Please write Solidity contracts with the below features, and add enough comments to demonstrate your design.
+
+This repo is scaffolded with basic tools, but you are free to use toolset of your choice.
+
+## Required features
+
+1. Owner of each notebook can write content into it, stored on-chain.
+2. Clients can retrieve content by the token id of notebook.
```

Comment:
+ `Notebook can receive donations and can be withdrawn by the owner.` for designing a safe contract?